bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -117,6 +118,7 @@
     ScheduledFuture<?> timeoutFuture = null;
 
     final long waitForWriteSetMs;
+    Map<Integer, BookieSocketAddress> delayedWriteFailedBookies = new HashMap<Integer, BookieSocketAddress>();
```

Comment:
Can you please explain why this is not  a ConcurrentMap ?

Reply:
Because of or ordered safe executor, I thought a regular map is good enough.

Reply:
I was thinking the same. Just wanted to double check.